packages/swing-store/src/snapStore.js

```diff
@@ -9,6 +9,45 @@ const pipe = promisify(pipeline);
 
 const { freeze } = Object;
 
+/**
+ *
+ * @param {import("fs").ReadStream | import("fs").WriteStream} stream
+ * @returns {Promise<void>}
+ */
+export const fsStreamReady = stream =>
+  new Promise((resolve, reject) => {
+    if (stream.destroyed) {
+      reject(new Error('Stream already destroyed'));
+      return;
+    }
+
+    if (!stream.pending) {
+      resolve();
+      return;
+    }
+
+    stream.destroyed;
+
+    const onReady = () => {
+      cleanup(); // eslint-disable-line no-use-before-define
+      resolve();
```

Comment:
is there any chance that an error would cause both `onReady` and `onError` to fire, causing a race between resolution and rejection?

Reply:
`onError` could happen after `onReady`, but at that point pipe should handle it. With the `cleanup` here, there is no way for both to be triggered at this step (and regardless, the resolve/reject already perform and already called check.